### PR TITLE
Minimal reference leak fixes

### PR DESCRIPTION
**Pull request by snmaynard**
I believe this is the minimal set of changes required to fix most of the reference leaks

### CODE REVIEW

File: src/BugsnagUnity/Native/Android/NativeInterface.cs [hunk 1]
```diff
@@ -279,9 +287,9 @@ public NativeInterface(AndroidJavaObject config)
       if (!isAttached) {
         AndroidJNI.AttachCurrentThread();
       }
-      using (var map = JavaMapFromDictionary(metadata))
+      using (AndroidJavaObject map = MakeJavaMapDisposable(metadata))
```
Comment (fractalwrench) [inline src/BugsnagUnity/Native/Android/NativeInterface.cs:290]: nit: consider renaming to `MakeDisposableJavaMap` as the current naming implies that `metadata` is of type `java/util/Map`

File: src/BugsnagUnity/Native/Android/NativeInterface.cs [hunk 2]
```diff
@@ -320,26 +331,45 @@ public List<Breadcrumb> GetBreadcrumbs()
...
 
       while (AndroidJNI.CallBooleanMethod(iterator, IteratorHasNext, new jvalue[]{}))
       {
-        var crumb = AndroidJNI.CallObjectMethod(iterator, IteratorNext, new jvalue[]{});
+        IntPtr crumb = AndroidJNI.CallObjectMethod(iterator, IteratorNext, new jvalue[]{});
         breadcrumbs.Add(ConvertToBreadcrumb(crumb));
         AndroidJNI.DeleteLocalRef(crumb);
       }
-
-      AndroidJNI.DeleteLocalRef(javaBreadcrumbs);
```
Comment (fractalwrench) [inline src/BugsnagUnity/Native/Android/NativeInterface.cs:334]: This local ref can still be deleted unless I'm mistaken?

Comment (snmaynard) [reply src/BugsnagUnity/Native/Android/NativeInterface.cs:334]: Yeah - I just moved the delete to being before the loop as we dont need a collection reference when using an iterator.

File: src/BugsnagUnity/Native/Android/NativeInterface.cs (hunk 1 above)
Comment (snmaynard) [reply src/BugsnagUnity/Native/Android/NativeInterface.cs:290]: This is the same format as MakeJavaString so its consistent. Maybe Build is better than Make?

File: src/BugsnagUnity/Native/Android/NativeInterface.cs (hunk 2 above)
Comment (fractalwrench) [reply src/BugsnagUnity/Native/Android/NativeInterface.cs:334]: ah, got it - that's fine.

File: src/BugsnagUnity/Native/Android/NativeInterface.cs (hunk 1 above)
Comment (fractalwrench) [reply src/BugsnagUnity/Native/Android/NativeInterface.cs:290]: Build sounds good to me

**Pull request merged**